File: frontend/olympic_it_project/lib/features/createExam/entities/question.dart
import 'answer.dart';
import 'difficulty.dart';
import 'question_type.dart';
import 'dart:io';



class Question {
  File? imageFile;
  File? videoFile;

  int order;

  QuestionType type;

  String content;

  String? imageUrl;

  String? videoUrl;

  Difficulty difficulty;

  List<Answer> answers;

  List<String> correctAnswerIds;

  Question({
    required this.order,
    required this.type,
    required this.content,
    this.imageFile,
    this.videoFile,
    required this.difficulty,
    this.answers = const [],
    this.correctAnswerIds = const [],
  });
}

File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/enity/Question.java
package org.example.olympic_ot_project.enity;

import jakarta.persistence.*;
import lombok.*;
import org.example.olympic_ot_project.core.QuestionLevel;
import org.example.olympic_ot_project.core.QuestionType;

import java.util.List;

@Entity
@Table(name = "question")
@Getter
@Setter
public class Question {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    @Enumerated(EnumType.STRING)
    private QuestionType type;

    @Enumerated(EnumType.STRING)
    private QuestionLevel level;

    @Column(columnDefinition = "TEXT")
    private String answer;

    @Column(nullable = false)
    private Integer score;

    @Column(name = "time_limit")
    private Integer timeLimit;

    private String imageUrl;

    private String videoUrl;

    @OneToMany(mappedBy = "question", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<QuestionOption> options;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}